### ISSUE TITLE
make boost fails

### ISSUE THREAD

**Issue by GoMrPickles** [victorliu/S4#86]
Hi, I'm trying to install this [fork](https://github.com/phoebe-p/S4), and was advised by @phoebe-p to open a thread here. I am seeing a failure when trying to compile S4 for use with Solcore. I am using Ubuntu 16.04, installing into an Anaconda Python 3.6 environment.

Steps:
```
sudo apt-get update
sudo apt install make
sudo apt install git
sudo apt install python3-pip
sudo pip3 install numpy matplotlib scipy
sudo apt-get install python3-tk
git clone https://github.com/phoebe-p/S4.git
sudo apt-get install libopenblas-dev
sudo apt-get install libfftw3-dev
sudo apt-get install libsuitesparse-dev

cd S4
make boost
make S4_pyext
```

Installation goes awry at the `make boost` step.

```make boost

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Boost dir is
tar -xzvf boost.tar.gz

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Makefile:110: recipe for target '/home/farmer/solcore5/S4/S4/lib' failed
make: *** [/home/farmer/solcore5/S4/S4/lib] Error 2
```
I'm not familar with `boost`, or frankly `make`, for that matter, outside cutting and pasting, so I'm not sure where to start with this error.

**Comment by kwrobert** [comment]
Hello there! 

Glad at least someone is attempting to use this fork :). Before you proceed, I think there is still a bug in calculating the z components of the H field that I haven't fixed yet. It might give inforrect results, thought I should throw that out there before I forget.

My best guess, without seeing whats in your directory, is that the boost.tar.gz file doesn't exist. I'm pretty sure I wrote the Makefile to automatically download and rename it, but that might not be working properly. Can you send me the output of the `tree` command at the root of the cloned repository?

**Comment by GoMrPickles** [comment]
[tree.txt](https://github.com/victorliu/S4/files/3507010/tree.txt)
Here you go, thanks!

**Comment by kwrobert** [comment]
Interesting, it appears the file is there. I think the problem is that for whatever reason the $(BOOST_DIR) variable that is computed dynamically in the Makefile after unpacking the archive of the boost source ended up empty. Perhaps the code in my Makefile is buggy. Can you send the output of `file boost.tar.gz` ? The file exists right at the top level of the S4 directory.

**Comment by GoMrPickles** [comment]
```
file boost.tar.gz
boost.tar.gz: empty
```

**Comment by kwrobert** [comment]
Aha, so that is the problem. For some reason the wget call in the Makefile isn't downloading the file properly. Do you have `wget` installed on your system? What is the output of `which wget`?

**Comment by GoMrPickles** [comment]
```
which wget
/usr/bin/wget
```
I have used `wget` on this system with no issues, recently.

**Comment by kwrobert** [comment]
Okay, can you run `wget https://sourceforge.net/projects/boost/files/boost/1.61.0/boost_1_61_0.tar.gz -O boost.tar.gz` from the top of the S4 directory.

**Comment by GoMrPickles** [comment]
```
wget https://sourceforge.net/projects/boost/files/boost/1.61.0/boost_1_61_0.tar.gz -O boost.tar.gz
--2019-08-15 14:41:13--  https://sourceforge.net/projects/boost/files/boost/1.61.0/boost_1_61_0.tar.gz
Resolving sourceforge.net (sourceforge.net)... 216.105.38.13
Connecting to sourceforge.net (sourceforge.net)|216.105.38.13|:443... connected.
Unable to establish SSL connection.
```
Odd... checking sourceforge.net with a web browser indicates a possible security cert issue?

**Comment by kwrobert** [comment]
Weird, I can hit that link no problem. Just tried it now and it starts downloading immediately. Looks like something might be up with your network. Isn't it funny how we set out to solve a science problem and, through a long and convoluted journey, end up debugging network issues?

**Comment by GoMrPickles** [comment]
Yes, especially after a day of on and off debugging. It looks like it may have to do with the firewall/blacklist setup at my office. I'll follow through with the IT folks and see if that resolves it. Thanks for your help stepping through this!

**Comment by kwrobert** [comment]
No problem. If you get those network issues resolved and something still doesn't work, feel free to open another issue. I'm going to close this for now to keep things tidy.

**Comment by kwrobert** [comment]
just kidding I can't close it because its against the original S4 repo and I don't have any privileges.

**Comment by m9631m** [comment]
> Hello there!
> 
> Glad at least someone is attempting to use this fork :). Before you proceed, I think there is still a bug in calculating the z components of the H field that I haven't fixed yet. It might give inforrect results, thought I should throw that out there before I forget.
> 
> My best guess, without seeing whats in your directory, is that the boost.tar.gz file doesn't exist. I'm pretty sure I wrote the Makefile to automatically download and rename it, but that might not be working properly. Can you send me the output of the `tree` command at the root of the cloned repository?

Can you explain more?

**Comment by kwrobert** [comment]
When you clone something from Github, you are cloning an entire repository containing a whole directory tree of files. So:

```
git clone https://github.com/kwrobert/S4.git
cd S4
```

brings you to the top/root of that directory tree. From there running the `tree` command gives you a printout showing the structure of that tree.

**Comment by m9631m** [comment]
Thanks for the reply. Sorry, my comment was unclear I meant can you explain about the bug in Z components?

" I think there is still a bug in calculating the z components of the H field that I haven't fixed yet. It might give inforrect results, thought I should throw that out there before I forget."

**Comment by kwrobert** [comment]
Ah yes, I found a bug in S4 that affected how it computes the z components of the e-field In dielectric materials that I have fixed. It appeared to be the result of copying and pasting code, just a misplaced line. Because the functions to compute the h and e fields are nearly identical, I suppose the same bug could also affect the z component of the h field, but I never found the time to test it on a magnetic material and find out
